src/private.h

```diff
@@ -74,6 +74,8 @@ typedef struct _partial_gpucomm {
 } partial_gpucomm;
 
 struct _gpuarray_buffer_ops {
+  int (*get_platform_count)(int* platcount);
+  int (*get_device_count)(int platform, int* devcount);
```

Comment:
Make the platform and device count an unsigned int because you can't have a negative count.